File: utils/src/main/java/uz/fido/utils/utility/language/LocaleHelper.java
package uz.fido.utils.utility.language;

import android.content.Context;
import android.content.res.Configuration;
import android.content.res.Resources;

import java.util.Locale;

import io.paperdb.Paper;

public class LocaleHelper {

    public static String getLanguage(Context context) {
        if (getPersistedData(Locale.getDefault().getLanguage()) != null && !getPersistedData(Locale.getDefault().getLanguage()).isEmpty()) {
            return getPersistedData(Locale.getDefault().getLanguage());
        } else {
            return "uz";
        }
    }

    public static Context setLocale(Context context, String language) {
        persist(language);
        return updateResources(context, language);
    }

    public static int getSelectedLang(Context context) {
        String str = getLanguage(context).toLowerCase(Locale.ROOT);
        switch (str) {
            case "en":
            case "eng":
                return 3;
            case "uz":
            case "uzl":
                return 2;
            case "uzc":
                return 1;
            default:
                return 0;
        }
    }

    private static String getPersistedData(String defaultLanguage) {
        return Paper.book().read("lang", defaultLanguage);
    }

    private static void persist(String language) {
        Paper.book().write("lang", language);
    }

    private static Context updateResources(Context context, String language) {
        Locale locale = new Locale(language);
        Locale.setDefault(locale);
        Resources res = context.getResources();
        Configuration config = new Configuration(res.getConfiguration());
        config.setLocale(locale);
        context = context.createConfigurationContext(config);
        return context;
    }
}

File: utils/src/main/java/uz/fido/utils/utility/language/LocaleHelper.kt
package uz.fido.utils.utility.language

import android.content.Context
import android.content.res.Configuration
import io.paperdb.Paper
import uz.fido.utils.const.Const.APP_LANGUAGE
import java.util.Locale

object LocaleHelper {

    fun getLanguage(): String {
        return if (getPersistedData(Locale.getDefault().language).isNotEmpty()) ({
            getPersistedData(Locale.getDefault().language)
        }).toString() else {
            "uz"
        }
    }

    fun setLocale(context: Context, language: String): Context {
        persist(language)
        return updateResources(context, language)
    }

    fun getSelectedLang(): Int {
        val lang = getLanguage().lowercase()
        return when (lang) {
            "en", "eng" -> 3
            "uz", "uzl" -> 2
            "uzc" -> 1
            else -> 0
        }
    }

    private fun getPersistedData(defaultLanguage: String): String {
        return Paper.book().read(APP_LANGUAGE, defaultLanguage)
    }

    private fun persist(language: String) {
        Paper.book().write(APP_LANGUAGE, language)
    }

    private fun updateResources(context: Context, language: String): Context {
        var context = context
        val locale = Locale(language)
        Locale.setDefault(locale)
        val res = context.resources
        val config = Configuration(res.configuration)
        config.setLocale(locale)
        context = context.createConfigurationContext(config)
        return context
    }
}